src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5954,10 +6007,18 @@ private static void ReportDuplicateObjectMemberInitializers(BoundExpression boun
 
                 var memberInitializerSyntax = boundMemberInitializer.Syntax;
 
-                Debug.Assert(memberInitializerSyntax.Kind() == SyntaxKind.SimpleAssignmentExpression);
-                var namedAssignment = (AssignmentExpressionSyntax)memberInitializerSyntax;
+                IdentifierNameSyntax memberNameSyntax;
```

Comment:
>IdentifierNameSyntax memberNameSyntax; [](http://example.com/codeflow?start=16&length=38)

Adjustments to this function likely won't be necessary if we create ```BoundAssignmentOperator``` as suggested because it will not pass the ```!boundMemberInitializer.HasAnyErrors``` test.

Reply:
Reverted